### PR TITLE
update pyproject.toml to current template format

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

When deprecating `ethPM`, I added `deprecation` to the valid newsfragment types in the newsfragments folder but not in `pyproject.toml'. 

Related to Issue #2953

### How was it fixed?

Set `pyproject.toml` to match the current template version, which includes `deprecation`. This required updating the newsfragment type `doc` to `docs` in a few places too.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/main/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,14 +1,15 @@
 [tool.towncrier]
-    package="web3"
-    filename = "docs/releases.rst"
-    directory = "newsfragments"
-    underlines = ["-", "~", "^"]
-    issue_format = "`#{issue} <https://github.com/ethereum/web3.py/issues/{issue}>`__"
-    title_format = "v{version} ({project_date})"
+# Read https://github.com/ethereum/<REPO_NAME>/blob/main/newsfragments/README.md for instructions
```
Comment (kclowes) [inline pyproject.toml]: nit:
```suggestion
# Read https://github.com/ethereum/web3.py/blob/main/newsfragments/README.md for instructions
```

**Review by kclowes — APPROVED** [review]
I left a nit, but LGTM other than that once tests are passing!

File: pyproject.toml (hunk 1 above)
Comment (pacrob) [reply pyproject.toml]: Yep, thank you :disappointed:

**Review by fselmo — APPROVED** [review]
Yep, lgtm after nit fix!

File: pyproject.toml [hunk 2]
```diff
@@ -1,14 +1,15 @@
 [tool.towncrier]
-    package="web3"
-    filename = "docs/releases.rst"
-    directory = "newsfragments"
-    underlines = ["-", "~", "^"]
-    issue_format = "`#{issue} <https://github.com/ethereum/web3.py/issues/{issue}>`__"
-    title_format = "v{version} ({project_date})"
+# Read https://github.com/ethereum/web3.py/blob/main/newsfragments/README.md for instructions
+package = "web3"
+filename = "docs/release_notes.rst"
```
Comment (fselmo) [inline pyproject.toml]: This should remain `releases.rst` at least until we apply the full template to `web3.py`

Comment (pacrob) [reply pyproject.toml]: set back to `releases.rst`

**Pull request merged**